src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -383,6 +386,150 @@ private static double[] CalculateAbsentFeatureLogProbabilities(int[] labelHistog
             return (ValueMapper<TIn, TOut>)(Delegate)del;
         }
 
+        /// <summary>
+        /// Creates an Onnx inferencing model by vectorizing and following the logic found in <see cref="Map"/>
+        /// </summary>
+        bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string featureColumn)
+        {
+            float[] featureHistogram = new float[_featureHistogram[0].Length * _labelHistogram.Length];
+            float[] labelHistogramExpanded = new float[_featureHistogram[0].Length * _labelHistogram.Length];
+
+            for (int i = 0; i < _featureHistogram.Length; i++)
+            {
+                Array.Copy(_featureHistogram[i], 0, featureHistogram, i * _featureHistogram[i].Length, _featureHistogram[i].Length);
+            }
+            for (int i = 0; i < _featureHistogram[0].Length; i++)
+            {
+                Array.Copy(_labelHistogram, 0, labelHistogramExpanded, i * _featureHistogram.Length, _featureHistogram.Length);
+            }
+
+            var one = ctx.AddInitializer((float)1, "one");
+            var zero = ctx.AddInitializer((float)0, "zero");
```

Comment:
A cleaner way of writing the same thing instead of casting is to use 1.0f. (The f suffix makes it a float)